src/transport/SessionManager.h

```diff
@@ -477,6 +500,12 @@ class DLL_EXPORT SessionManager : public TransportMgrDelegate, public FabricTabl
     State mState; // < Initialization state of the object
     chip::Transport::GroupOutgoingCounters mGroupClientCounter;
 
+#if INET_CONFIG_ENABLE_TCP_ENDPOINT
+    OnConnectionCompleteCallback mConnCompleteCb = nullptr;
+    OnConnectionClosedCallback mConnClosedCb     = nullptr;
+    void * mConnContext                          = nullptr;
```

Comment:
This makes no sense, in that CASESession sets this when sending Sigma1.  There's one SessionManager, but there can be (and are, for real clients like hubs) many parallel CASE establishments happening at once.

Put another way: when a hub boots, it will in parallel try to do CASE with all the devices.  That will totally break here.

Reply:
This was an oversight. The SessionManager needs to handle multiple sessions. One method would be to pass an appContext down and get it back on connection completion to be used to multiplex among the various sessions at the SessionManager.

Reply:
Created an App context that is passed in with ConnectToPeer() and plumbed all the way to the Connection context object and returned on connection completion to get to the right CASESession.